packages/contracts/contracts/core/plugin/PluginTransparentUpgradeable.sol

```diff
@@ -0,0 +1,42 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.10;
+
+import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import "@openzeppelin/contracts-upgradeable/utils/introspection/ERC165Upgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/proxy/utils/UUPSUpgradeable.sol";
+
+import {PermissionManager} from "../permission/PermissionManager.sol";
+
+import {AppStorage} from "../../utils/AppStorage.sol";
+import {IDAO} from "../IDAO.sol";
+import {DaoAuthorizableUpgradeable} from "../component/DaoAuthorizableUpgradeable.sol";
+
+/// @title PluginTranparentUpgradable
+/// @notice An abstract contract to inherit from when creating a proxy contract.
+/// This should be used to deploy logic contracts where proxy itself
+/// is deployed through transparent or beacon...
+abstract contract PluginTransparentUpgradeable is
+    Initializable,
+    ERC165Upgradeable,
+    DaoAuthorizableUpgradeable
+{
+    bytes4 public constant PLUGIN_INTERFACE_ID = type(PluginTransparentUpgradeable).interfaceId;
+
+    function __Plugin_init(address _dao) internal virtual onlyInitializing {
```

Comment:
How do we make sure that the implementation contract can't be initialized again ? 

OZ writes the following (https://docs.openzeppelin.com/contracts/4.x/api/proxy#Initializable)
> Avoid leaving a contract uninitialized.
> An uninitialized contract can be taken over by an attacker. This applies to both a proxy and its implementation contract, which may impact the proxy. To prevent the implementation contract from being used, you should invoke the [_disableInitializers](https://docs.openzeppelin.com/contracts/4.x/api/proxy#Initializable-_disableInitializers--) function in the constructor to automatically lock it when it is deployed:
```Solidity
/// @custom:oz-upgrades-unsafe-allow constructor
constructor() {
    _disableInitializers();
}
```